python/tank/descriptor/io_descriptor/git_tag.py

```diff
@@ -311,8 +204,53 @@ def get_latest_version(self, constraint_pattern=None):
 
         :returns: IODescriptorGitTag object
         """
-        if constraint_pattern:
-            return self._get_latest_by_pattern(constraint_pattern)
-        else:
-            return self._get_latest_version()
+        # figure out the latest commit for the given repo and branch
+        # 'git ls-remote --tags repo_url' returns
+        #
+        # 32862181fa982aba99af1e518c51117309009023	refs/tags/v0.18.5rv
+        # f08d7e0e5f5bf1402d21a653510c8be77b184f93	refs/tags/v0.18.5rv^{}
+        # 5c612bd01cc0db46408e715aca360a133d4eefeb	refs/tags/v0.18.6
+        # 4ecc15ac7045eef0d5ab205dede3c5c5f513ec09	refs/tags/v0.18.6^{}
+        # 6519356b7f4a829205a1cb71be9ec5b5e4e10409	refs/tags/v0.18.7
+        # a8e5ceaa4222b0d8b35cff0a0d2ef1bd680c1a1e	refs/tags/v0.18.7^{}
+        # 3c9494743cf7c8e07806b1ae9f1af152994c1fe6	refs/tags/v0.18.8
+        # b92f3cc69060b73c26e6236ef2c65859794f1f4e	refs/tags/v0.18.8^{}
+        # 5e7014ca28e94f7baf2cea5ca9ba103414095897	refs/tags/v0.18.9
+        # 17041e87070354a7c221e33d5a4edd02fd0159a6	refs/tags/v0.18.9^{}
+        #
+        try:
+            command = "ls-remote --tags \"%s\"" % self._sanitized_repo_path
+            all_tag_chunks = execute_git_command(command).split("\n")
+            log.debug("ls-remote returned: '%s'" % all_tag_chunks)
+
+            # get first chunk of return data
+            tags = []
+            for tag_chunk_line in all_tag_chunks:
+                # 4ecc15ac7045eef0d5ab205dede3c5c5f513ec09	refs/tags/v0.18.6^{}
+                tag_match = re.match("^.*refs/tags/([^\^]+)$", tag_chunk_line)
+                if tag_match:
+                    tags.append(tag_match.group(1))
+
+            if len(tags) == 0:
+                raise TankDescriptorError("No tags defined!")
+
+            if constraint_pattern is None:
+                # get latest tag
+                version_to_use = tags[-1]
```

Comment:
This is not true. If you had run this on core, v0.18.9 would be the latest tag. Maybe using LooseVersion for comparison?